dttools/src/text_list.c

```diff
@@ -75,4 +76,27 @@ void text_list_delete(struct text_list *t)
 	free(t);
 }
 
+struct text_list *text_list_allpairs_remote_create(const char *path, const char *set)
```

Comment:
I'm going to nitpick here.  `text_list` is a general purpose data structure, so it should include any application specific code.  This function is very specific to the allpairs tools, so it should be a module in the allpairs directory and invoke `text_list_create/append` externally.